File: algorithms-step-by-step/recursion/exercises/recursiveRange.py
def recursiveRange(number):
    result = 0
    index = 0

    def add_recursive_range():
        nonlocal index
        nonlocal result
        if index > number:
            return
        result = result + index
        index = index + 1
        add_recursive_range()
    add_recursive_range()
    return result


print(recursiveRange(100))

File: algorithms-step-by-step/recursion/exercises/recursiveRange.js
function recursiveRange(number) {
  let result = 0;
  let index = 0;
  function addRecursiveRange() {
    if (index > number) {
      return;
    }
    result += index;
    index++;
    addRecursiveRange();
  }
  addRecursiveRange();
  return result;
}

console.log(recursiveRange(10));
